### PR TITLE
Flatten Scheduler Struct

### LINKED ISSUE #71860: [scheduler cleanup phase 2]: Flatten Top Level Scheduler Struct
**PLEASE COORDINATE WITH SIG-SCHEDULING BEFORE WORKING ON THIS ISSUE.** We want to avoid the situation where two people accidentally start work on the same fix.

Main tracking issue: https://github.com/kubernetes/kubernetes/issues/68951

If you contributed in phase 1, please give others a chance before claiming this issue. You can also help out by reviewing the phase 2 PRs.

Currently, the type [`scheduler.Scheduler`](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go) is a struct with one field. (`factory.Config`) It is misleading for all of the scheduler’s config *and state* to be stored in a type called "config".

Instead, we can merge these two types, keeping only `scheduler.Scheduler`. Any function returning a Config will return a Scheduler instead. Integration tests will need to be fixed to accommodate this change.

/sig scheduling
/kind cleanup
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by ahmad-diaa**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The type `scheduler.Scheduler` is a struct with one field (factory.Config). It is misleading for all of the scheduler’s config and state to be stored in a type called "config". This PR flattens the scheduler struct. Also, this will later help us to make the scheduler independent of the `factory.Configurator` type.

**Which issue(s) this PR fixes**:

Fixes #71860

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test 
/priority important-longterm

File: test/integration/volumescheduling/util.go [hunk 1]
```diff
@@ -123,6 +123,13 @@ func initTestSchedulerWithOptions(
 
 	// set DisablePreemption option
 	context.schedulerConfig.DisablePreemption = false
+	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{
```
Comment (draveness) [inline test/integration/volumescheduling/util.go]: why are you moving these lines?

**Review by draveness — CHANGES_REQUESTED** [review]

File: test/integration/volumescheduling/util.go (hunk 1 above)
Comment (ahmad-diaa) [reply test/integration/volumescheduling/util.go]: initializing the recorder after the call to `NewFromConfig`  didn't update in `sched.Recorder`. Any better idea to do this?

Comment (draveness) [reply test/integration/volumescheduling/util.go]: SG

**Review by draveness — COMMENTED** [review]
The code generally looks good, could you please squash the commits?

**Review by draveness — COMMENTED** [review]
/lgtm

Thanks @ahmad-diaa

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks, @ahmad-diaa !

Due to recent refactoring of volume binding integration tests, needs @msau42 to /approve on that part.

**Comment by alculquicondor** [comment]
Feel free to squash at this point :)

**Pull request merged**